Clarify that `ok_button_text` does not work reliably in `FileDialog`

AThousandShips opened the pull request:
Hide the `ok_button_text` property
Also hide the `title` property if generated by mode

Got an alternative solution, outlined in the issue, and can change this to use that solution if desired, but starting out with a more basic approach, if wanted I can merge that process into this as a separate commit for testing as well

* Fixes: #79803 

Can be retargeted for 4.1 if superseded, assuming we don't want to break compat for 4.1
<!--
Please target the `master` branch in priority.
PRs can target `3.x` if the same change was done in `master`, or is not relevant there.

Relevant fixes are cherry-picked for stable branches as needed by maintainers.
You can mention in the description if the change is compatible with `3.x`.

To speed up the contribution process and avoid CI errors, please set up pre-commit hooks locally:
https://docs.godotengine.org/en/latest/contributing/development/code_style_guidelines.html
-->

AThousandShips commented:
Can also do the same for `EditorFileDialog` if wanted, as it is documented

KoBeWi commented:
I think dialogs should have something like "internal button text", which is used when the property is empty. Modifying the property automatically brings many problems, including properties being translated and saved in the scene.
(I was planning to do something about it at some point)

AThousandShips commented:
With this the problems of saving goes away, but those are good points, will look at that as yet another option

AThousandShips committed:
Clarify that `ok_button_text` does not work reliably in `FileDialog`

Hide the `ok_button_text` property
Also hide the `title` property if generated by mode
